crux-core/src/crux/tx.clj

```diff
@@ -99,13 +99,14 @@
                                         (c/->EntityTx eid end-valid-time transact-time tx-id (c/nil-id-buffer)))])))
 
                          (->> (cons start-valid-time
-                                    (with-open [entity-as-of-idx (idx/new-entity-as-of-index snapshot start-valid-time transact-time)]
-                                      (when-let [visible-entity (some-> (idx/entity-at entity-as-of-idx eid)
-                                                                        (select-keys [:tx-time :tx-id :content-hash]))]
-                                        (->> (idx/entity-history-seq-ascending snapshot eid start-valid-time transact-time)
-                                             (remove #{start-valid-time})
-                                             (take-while #(= visible-entity (select-keys % [:tx-time :tx-id :content-hash])))
-                                             (map #(.vt ^EntityTx %))))))
+                                    (with-open [i (kv/new-iterator snapshot)]
```

Comment:
Similar, as this returns a lazy sequence you want to realise the sequence before the with-open block ends. Using with-open in this case is fine though. Worth noting that this issue was already there.

Reply:
hmm, interesting that it passed all the tests then :grimacing: anyway, mapv'd, cheers